src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryModuleConfigurator.cs

```diff
@@ -0,0 +1,43 @@
+﻿namespace Microsoft.ApplicationInsights.AspNetCore
+{
+    using System;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Represents method used to configure <see cref="ITelemetryModule"/> with dependency injection support.
+    /// </summary>
+    internal class TelemetryModuleConfigurator : ITelemetryModuleConfigurator
+    {
+        private Action<ITelemetryModule> configure;
+        internal readonly Type telemetryModuleType;
+
+        /// <summary>
+        /// Constructs an instance of <see cref="TelemetryModuleConfigurator"/>.
+        /// </summary>        
+        /// <param name="configure">The action used to configure telemetry module.</param>
+        /// <param name="telemetryModuleType">The type of the telemetry module being configured.</param>
+        public TelemetryModuleConfigurator(Action<ITelemetryModule> configure, Type telemetryModuleType)
```

Comment:
Null check for `configure`.

Reply:
nullcheck done before calling Invoke. Line 32